### ISSUE TITLE
Add dragonfly commands for loading and recognising from command modules

### ISSUE THREAD

**Issue by Danesprite** [dictation-toolbox/dragonfly#98]
This would be a new command for dragonfly's command-line interface (see the [CLI documentation](https://dragonfly2.readthedocs.io/en/latest/cli.html)). I guess the command would be `dragonfly load <files>`.

It would be similar to dragonfly's `test <files>` command, but would call the relevant engine method to recognise speech from the microphone. Most of the arguments for `test` could be re-used, e.g. `-e|--engine` and `--language`. 

It would be an alternative to the module loaders and should work with each available engine. The observer used by the module loaders could be used with this command to print `"Speech started."`, `"Recognized: <words>"` and `"Sorry, what was that?"`. This and other parts of the module loaders could be moved into `dragonfly/__main__.py` or `dragonfly/loader.py` to avoid duplicating code.

A `-c|--config` option could be added for loading a Python module before the others. The config module could be used for engine configuration and setting up other functionality like the wake/sleep modes added by @daanzu's module loaders for Kaldi and WSR.

One benefit of this command over the module loaders is greater control over which modules are loaded. You could specify modules directly or with shell wildcards instead of changing the loader code. For example:
```Shell
python -m dragonfly load _module1.py _module2.py
python -m dragonfly load *.py
python -m dragonfly load _*.py
```

This already works with `dragonfly test`.

**Comment by LexiconCode** [comment]
An adjunctive to the CLI could be that dragonfly has a configuration file that defining engine loading module and path to directory containing dragonfly scripts. That way a third-party program such as Caster could create the configuration file or reconfigure programmatically.

**Comment by Danesprite** [comment]
I can see where you're coming from, but I think it would be easier to add another command that takes a directory path instead of multiple command module paths like the `test` command currently does. For example:

```shell
python -m dragonfly load-directory "C:\Users\Default User\dragonfly-modules"

```

This command could also have options like `-e/--engine`, `-l/--language`,`-q/--quiet` and `-c/--config`.

**Comment by daanzu** [comment]
@Danesprite This seems like a nice idea. I wonder if there is a simple enough way to integrate it with the module loaders, so that it calls them (using CLI args or directly in python) to do the loading/etc, calling the appropriate one based on the requested engine. That would mean less code duplication. It would make the examples more complex, but any one needing to modify them is probably knowledgeable enough to deal with that.

**Comment by Danesprite** [comment]
It should be easy enough to change the module loaders to call functions that implement my proposed CLI commands. The `get_engine()` calls would be left in-tact before the CLI command function calls so that optional engine arguments can still be specified. The `-c/--config` argument could be used to pass optional arguments to `get_engine()` if using the CLI command. 

With that in mind, perhaps an option for disabling recognition state feedback messages would also be useful in case a user wants to customise them:

```Shell
python -m dragonfly load --no-recobs-messages observers.py grammar1.py
python -m dragonfly load-directory --no-recobs-messages "C:\Users\Default User\dragonfly-modules"

```

The current observer functions and that command-line option could be used in module loaders so that it's easier to change the observer messages without digging into the docs.

It should also be possible to allow passing optional command-line arguments directly to `get_engine()` now that I think about it. For example, with the Kaldi engine options:

```Shell
python -m dragonfly load-directory -o model_dir="C:\Users\Default User\kaldi_model" -o vad_padding_end_ms=300 "C:\Users\Default User\dragonfly-modules"

```

Not sure how difficult that is with `argparse` though.

**Comment by LexiconCode** [comment]
Something else to consider here is to have more than one module loader active.
- My primary thought is for [noise recognition](https://github.com/daanzu/dragonfly_daanzu_tools) module loaders

**Comment by daanzu** [comment]
Actually, my noise recognition module shouldn't need a separate loader. It should work just by putting the file in your usual dragonfly directory.

**Comment by Danesprite** [comment]
Most of this is implemented in PR #191 now.

`argparse` doesn't seem to like multiple optional arguments like `-o model_dir=. -o vad_padding_end_ms=300`, so I opted for separating the `argument=value` parts with commas instead. It won't work for all arguments yet though, e.g. the VAD argument will cause a type error during engine initialization.

I don't think a `-c/--config` argument would be easy to implement nicely. It should be good enough to pass arguments to `get_engine()`, although maybe I'm missing some cases where that's not the case.

Multiple module loaders can work by just separately running multiple commands. If the use case was using both for speech recognition, then the tricky part would be to have mutually exclusive contexts between running engines, but I think this is something that should be solved externally instead, not within dragonfly.
